chia/server/node_discovery.py

```diff
@@ -291,13 +299,16 @@ async def _connect_to_peers(self, random: Random) -> None:
                         await asyncio.sleep(introducer_backoff)
                     except asyncio.CancelledError:
                         return None
-                    # Run dual between DNS servers and introducers. One time query DNS server,
-                    # next two times query the introducer.
-                    if introducer_attempts % 3 == 0 and len(self.dns_servers) > 0:
+                    # Alternate between DNS servers and introducers.
+                    # First try all the DNS servers in the list once. Then try the introducers once.
+                    if len(self.dns_servers) > 0 and not tried_all_dns_servers:
                         dns_address = self.dns_servers[dns_server_index]
-                        dns_server_index = (dns_server_index + 1) % len(self.dns_servers)
+                        next_index = (dns_server_index + 1) % len(self.dns_servers)
```

Comment:
`next_index` seems redundant. it looks like you could just increment `dns_server_index` directly

Reply:
True. I just thought it was more clear naming it, to clarify that we intended to use the old value, not the new value.

Reply:
(That is, use the old value to index `dns_address`)